llvm/include/llvm/Support/Parallel.h

```diff
@@ -96,9 +96,8 @@ class TaskGroup {
 
   // Spawn a task, but does not wait for it to finish.
   // Tasks marked with \p Sequential will be executed
-  // exactly in the order which they were spawned.
-  // Note: Sequential tasks may be executed on different
-  // threads, but strictly in sequential order.
+  // exactly in the order which they were spawned and
+  // on the thread with index 0.
```

Comment:
I remember this behaviour being slightly concerning when this functionality was added and now I know why...

Reply:
yep.

Reply:
FTR, the discussion about the "concerning"  behavior is this comment ("I don't think it is a problem as far as correctness, but might be somewhat unexpected.") on https://reviews.llvm.org/D148728